OSDOCS#7528: Reorganizing namespace exclusions and disabling after mo…

bergerhoffer opened the pull request:
…difying

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
4.14+

Issue:
https://issues.redhat.com/browse/OSDOCS-7528

Link to docs preview:
https://64343--docspreview.netlify.app/openshift-enterprise/latest/authentication/understanding-and-managing-pod-security-admission#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

bergerhoffer commented:
@stlaz Can you please review? In addition to noting that sync is disabled if users modify the PSA labels, I've also moved the info on namespaces with sync permanently (and initially) disabled into its own section instead of all that squeezed into a note. Let me know what you think.

https://64343--docspreview.netlify.app/openshift-enterprise/latest/authentication/understanding-and-managing-pod-security-admission#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

stlaz commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,41 @@
+[id="security-context-constraints-psa-sync-exclusions_{context}"]
+= Pod security admission synchronization namespace exclusions
+
+Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.
```
```suggestion
Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled on user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.
```

stlaz commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,41 @@
+
+[IMPORTANT]
+====
+If the pod security admission labels (`pod-security.kubernetes.io/<mode>`) are manually modified away from the system defaults on a namespace, synchronization is disabled for that namespace. Users can enable synchronization again, if necessary.
```
See if the following makes sense
```suggestion
If a pod security admission label (`pod-security.kubernetes.io/<mode>`) is manually modified from the auto-labelled value on a label-synchronized namespace, synchronization is disabled for that label. Users can enable synchronization again, if necessary, by removing the label from the namespace or by forcing synchronization by setting the `security.openshift.io/scc.podSecurityLabelSync` label to `true`.
```

stlaz commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,41 @@
+You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`.
+====
+
+If an Operator is installed in a user-created `openshift-*` namespace, synchronization enabled automatically after a cluster service version (CSV) is created in the namespace. The synchronized label inherits the permissions of the service accounts in the namespace.
```
```suggestion
If an Operator is installed in a user-created `openshift-*` namespace, synchronization is enabled automatically after a cluster service version (CSV) is created in the namespace. The synchronized label inherits the permissions of the service accounts in the namespace.
```

"the label inherits permissions" sounds odd, should we say that it's "derived from the permissions"?

bergerhoffer commented:
@stlaz I updated per your feedback and added a note about what happens if you force sync. Let me know what you think, thanks!

https://64343--docspreview.netlify.app/openshift-enterprise/latest/authentication/understanding-and-managing-pod-security-admission#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

bergerhoffer commented:
@stlaz Can you PTAL?

stlaz commented:
/lgtm

bergerhoffer commented:
@gkarager Can you PTAL? In addition to noting that sync is disabled if users modify the PSA labels, I've also moved the info on namespaces with sync permanently (and initially) disabled into its own section instead of all that squeezed into a note. 

Preview: https://64343--docspreview.netlify.app/openshift-enterprise/latest/authentication/understanding-and-managing-pod-security-admission#security-context-constraints-psa-synchronization_understanding-and-managing-pod-security-admission

bergerhoffer commented:
/label peer-review-needed

bergerhoffer commented:
Peer reviewer: note that some of this content is just moved from one file to another

jeana-redhat commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,46 @@
+[discrete]
+== Initially disabled namespaces
+
+By default, all namespaces that have an `openshift-` prefix have pod security admission synchronization disabled initially. You can enable synchronization for user-created [x-]`openshift-*` namespaces and for `openshift-operators`.
```
```suggestion
By default, all namespaces that have an `openshift-` prefix have pod security admission synchronization disabled initially. You can enable synchronization for user-created [x-]`openshift-*` namespaces and for the `openshift-operators` namespace.
```

jeana-redhat commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,46 @@
+
+[NOTE]
+====
+You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`.
```
```suggestion
You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for the `openshift-operators` namespace.
```

As an aside, what does `[x-]` do here?

bergerhoffer replied:
Sometimes when you have an asterisk inside backticks, it has issues in thinking you're trying to bold. I can't remember the exact scenarios when issues happens, but it sometimes fails on Travis builds and sometimes will bold text afterward that you didn't mean to. This `[x-]` escapes that so that it doesn't try to process the * as bold.

I can't tell for sure if it's needed in this case, but since it was there in the text I copied from, it's safest to leave it in.

jeana-redhat reviewed (commented):
Just a couple things to think about from me, looks good!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

gkarager commented on modules/security-context-constraints-psa-opting.adoc:
```diff
@@ -10,18 +10,7 @@ You can enable or disable automatic pod security admission synchronization for m
-By default, all namespaces that have an `openshift-` prefix are not synchronized. You can enable synchronization for any user-created [x-]`openshift-*` namespaces. You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`. 
-
-If an Operator is installed in a user-created `openshift-*` namespace, synchronization is turned on by default after a cluster service version (CSV) is created in the namespace. The synchronized label inherits the permissions of the service accounts in the namespace.
+You cannot enable pod security admission synchronization on some system-created namespaces. For more information, see _Pod security admission synchronization exclusions_.
```
@bergerhoffer Is the link for "Pod security admission synchronization exclusions." missing here or it is supposed to be just a text?

bergerhoffer replied:
@gkarager It is supposed to be just text.

Unfortunately, we can't put cross-references to other areas of our doc set in modules (anything in the modules/ directory). If we need to link, we usually add a link right after the module is included, in an additional resources section.

Doesn't look like I did that this time, maybe because it's so close to it above. But it looks like I changed the name in the section, but didn't here, so that's not helpful here. I will add a link right after the module here.

bergerhoffer commented on modules/security-context-constraints-psa-sync-exclusions.adoc:
```diff
@@ -0,0 +1,46 @@
+
+Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled on user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.
+
+[IMPORTANT]
```
Yeah I played with this a few different ways, and wasn't super happy with any of them. This was the best way I could come up with, but I'm open to other ideas if you have them

jeana-redhat replied:
I also couldn't think of anything particularly helpful... maybe next time this module needs an edit something brilliant will occur :joy:

gkarager commented:
/lgtm

bergerhoffer commented:
/cherrypick enterprise-4.14
